### DISCUSSION TITLE
Extração das NFs (NFe + NFCe) emitidas contra o CNPJ da empresa

### DISCUSSION THREAD

**Discussion by rodovalho** [TadaSoftware/PyNFe#286] [Q&A] [answered]
Olá,

Estou precisando implementar um método para a extração em lote, possivelmente de forma diária, de todas as NFs emitidas contra um CNPJ específico, ou seja, as NFs nas quais o citado CNPJ seja o beneficiário da NF e não o emitente.

O objetivo seria duplo. O primeiro, mais óbvio, seria o de automatizar o controle das compras/fornecedores. O segundo é o de monitorar eventuais NFs emitidas de forma indevida para permitir subsequente manifestação junto a SEFAZ.

Estive tentando usa o método descrito no tópico "Fluxo da Consulta de Distribuicao por NSU", porém, os resultados tem vindo zerados. 
```
Nova consulta a partir do NSU: 000000000000000
Quantidade de NSUs na consulta atual: 0
cStat: 137
xMotivo: Nenhum documento localizado
maxNSU: 000000000000000
Não há mais documentos a pesquisar
```

Gostaria de saber se esse seria o caminho apropriado. Talvez esteja cometando algum erro, já que existem diversas NFe e NFCe já emitidas para o CNPJ em questão como o destinatário. 

Desde já agradeço a ajuda.

**Comment by aron0306** [comment]
@rodovalho Já se passou muito tempo, mas não custa responder. Creio que precise encontrar uma forma diferente para fazer a consulta que deseja. Esta consulta por NSU tem limitações bem rígidas que estão te bloqueando.
Os limites estão descritos neste informe [https://www.nfe.fazenda.gov.br/portal/informe.aspx?ehCTG=false&Informe=0cu/yBLKrCs=](url)
Como pode ver no ponto 1.1, este tipo de consulta por NSU só serve para consultar a nota mais recente e quando não há novas notas você deve parar de consultar por 1h ou será bloqueado por 1h e a cada nova consulta antes dessa 1h, o tempo de bloqueio será reiniciado.
Se descobriu ou vier a descobrir como fazer a consulta retroativo em lote por favor me avise pois tenho o mesmo interesse que você.

**Comment by talesmm14** [comment]
Alguma atualização? Estou com o mesmo problema.

**Reply by aron0306** [reply]
Cara, fóruns aqui tão abandonados...
Vou ver se contrato algum dev com experiência na área pra implementar o que preciso.

**Comment by leogregianin** [comment]
@rodovalho @aron0306 @talesmm14 
Primeiro de tudo, recomendo ler todas as notas técnicas sobre o assunto e entender como o webservice funciona antes de começar a implementar algo.
NFCe não existe no Webservice de Distribuição de DFe, nem no Portal Nacional existem.
Se o retorno da consulta da Distribuição não está vindo nenhuma nota, quer dizer que você não fez o processo correto em Consultar, Manifestar e depois Baixar. 
Notas retroativas não são disponibilizadas pelo webservice. Muitos sistemas fazem scraping no [Portal da NFe](https://www.nfe.fazenda.gov.br/portal/principal.aspx) onde existem prós e contras.

**Reply by EduardoFerreira22** [reply]
Olá, @leogregianin!

Achei muito interessante oque vc comentou. Gostaria de saber se vc poderia explicar mais sobre, e se seria possível também buscar NFS-e pelo Pela distribuição DFe?

Desde já agradeço!

**Reply by juniortada** [reply, marked as answer]
@EduardoFerreira22 NFS-e não existe no Webservice de Distribuição de DFe, somente NF-e.

**Reply by EduardoFerreira22** [reply]
Obrigado!
